Application/Business/UserServices/UserServices.cs

```diff
@@ -25,39 +28,53 @@ public async Task<int> Count()
     }
     public async Task Create(UserEntity user) {
         var userList = List();
-        var check = false;
         foreach(var User in userList.Result)
         {
             if(User.Email == user.Email)
             {
-                check = true;
-                break;
+                throw new BadRequestException("Email exists");
             }
         }
-        if(!check)
+        if(Validation(user) == false)
         {
-            var newUser = new UserEntity
-            {
-                Username = user.Username,
-                Password = user.Password,
-                FirstName = user.FirstName,
-                LastName = user.LastName,
-                Avatar = user.Avatar,
-                Email = user.Email,
-                Phone = user.Phone,
-                Code = user.Code,
-                Status = user.Status,
-                Balance = user.Balance
-            };
-            await _userRepo.CreateAsync(user);
+            throw new BadRequestException("Email/phone not in correct format");
         }
+        await _userRepo.CreateAsync(user);
     }
     public async Task Update(Guid UserId, UserEntity user) {
         var target = await GetById(UserId);
-        if(target is not null)
+        if(target is null)
+        {
+            throw new NotFoundException("User not exist");
+        }
+        if (Validation(user) == false)
+        {
+            throw new BadRequestException("Email/phone not in correct format");
+        }
+        await _userRepo.UpdateAsync(target);
+    }
+    public async Task Delete(Guid UserId) {
+        var target = await _userRepo.DeleteSoftAsync(UserId);
+        if(target is null)
+        {
+            throw new NotFoundException("User not exist");
+        }
+    }
+    public bool Validation(UserEntity user)
+    {
+        var valid = true;
+        try
+        {
+            var emailAddress = new MailAddress(user.Email);
```

Comment:
we can implement with DataAnotation, or implement like this [link](https://stackoverflow.com/questions/16100300/asp-net-mvc-custom-validation-by-dataannotation) to make our business class more cleaner